menas/src/main/scala/za/co/absa/enceladus/menas/services/OozieService.scala

```diff
@@ -15,20 +15,50 @@
 
 package za.co.absa.enceladus.menas.services
 
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
 import scala.concurrent.Future
 
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
 
+import za.co.absa.enceladus.menas.exceptions.OozieActionException
 import za.co.absa.enceladus.menas.models.OozieCoordinatorStatus
 import za.co.absa.enceladus.menas.repositories.OozieRepository
+import za.co.absa.enceladus.model.menas.scheduler.oozie.OozieSchedule
 
 @Component
 class OozieService @Autowired() (oozieRepository: OozieRepository) {
 
+  import scala.concurrent.ExecutionContext.Implicits.global
+
   def isOozieEnabled: Boolean = oozieRepository.isOozieEnabled()
 
   def getCoordinatorStatus(coordinatorId: String): Future[OozieCoordinatorStatus] = {
     oozieRepository.getCoordinatorStatus(coordinatorId)
   }
+
+  def runNow(oozieSchedule: OozieSchedule, reportDate: Option[String]): Future[String] = {
+    val wfPath = oozieSchedule.activeInstance match {
+      case Some(instance) => instance.workflowPath
+      case None           => throw OozieActionException("Cannot run a job without an active schedule. Click 'Edit Schedule' to configure schedule first.")
```

Comment:
This is a response from a REST API, shouldn't mention or even know about any UI things like clicking buttons.

Reply:
Makes sense